rhino/src/main/java/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -135,21 +135,39 @@ public abstract class ScriptableObject
 
     protected static ScriptableObject buildDataDescriptor(
             Scriptable scope, Object value, int attributes) {
+        return buildDataDescriptor(scope, value, attributes, null, -1);
+    }
+
+    protected static ScriptableObject buildDataDescriptor(
+            Scriptable scope, Object value, int attributes, String name, int length) {
         ScriptableObject desc = new NativeObject();
         ScriptRuntime.setBuiltinProtoAndParent(desc, scope, TopLevel.Builtins.Object);
         desc.defineProperty("value", value, EMPTY);
-        desc.setCommonDescriptorProperties(attributes, true);
+        desc.setCommonDescriptorProperties(attributes, true, name, length);
```

Comment:
is it a good idea to introduce this new function with two (at least for me magic parameters) including all the forwarder for the old calls if we need the additional functionality only at one place?

Reply:
Removed these as they don't actually solve the problem with name and length attributes.